libfwupd/fwupd-converter.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#pragma once
+
+#include <gio/gio.h>
+#include <json-glib/json-glib.h>
+
+#include "fwupd-build.h"
+
+#define FWUPD_TYPE_CONVERTER (fwupd_converter_get_type())
+G_DECLARE_INTERFACE(FwupdConverter, fwupd_converter, FWUPD, CONVERTER, GObject)
```

Comment:
@superm1 better name welcome for this btw -- it's really serializing and deserializing the object, rather than _converting_ it...

Reply:
`FwupdSerializer`?

Reply:
I need something ideally less long -- and it needs to convey it's able to go both ways.

Reply:
`FwupdTransform`? `FwupdMeta`? `FwupdFormat`? `FwupdFmt`? `FwupdModel`?

Reply:
`FwupdDataTx`?

Reply:
FwupdCodec perhaps? Also FwupdFmt is nice because of the rust Fmt stuff.

Reply:
I like codec! Encoding and decoding.